Sources/SnapshotTesting/Common/ForwardingWKNavigationDelegate.swift

```diff
@@ -0,0 +1,133 @@
+#if os(macOS)
+import Cocoa
+#endif
+#if os(iOS)
+import UIKit
+#endif
+
+#if os(iOS) || os(macOS)
+import WebKit
+
+final class ForwardingWKNavigationDelegate: NSObject {
+  var didFinish: (() -> Void)?
+  var original: WKNavigationDelegate?
+  init(didFinish: @escaping () -> Void = {}, originalDelegate: WKNavigationDelegate?) {
+    self.original = originalDelegate
+    self.didFinish = didFinish
+  }
+
+  override public func responds(to aSelector: Selector!) -> Bool {
+    if decidePolicyPre13 == aSelector, original?.responds(to: aSelector) != true {
+      return NSObject.responds(to: aSelector)
+    }
+
+    if
+      #available(iOSApplicationExtension 13.0, *, OSXApplicationExtension 10.15, *),
+      decidePolicy13 == aSelector,
+      original?.responds(to: aSelector) != true
+    {
+      return NSObject.responds(to: aSelector)
+    }
+
+    return interceptedSelectors.contains(aSelector)
+      || original?.responds(to: aSelector) == true
+      || NSObject.responds(to: aSelector)
+  }
+
+  override public func forwardingTarget(for aSelector: Selector!) -> Any? {
+    interceptedSelectors.contains(aSelector)
+      ? nil
+      : original
+  }
+
+  func finish() {
+    didFinish?()
+    didFinish = nil
+  }
+}
+
+extension ForwardingWKNavigationDelegate: WKNavigationDelegate {
+  @available(iOSApplicationExtension 13.0, *, OSXApplicationExtension 10.15, *)
+  func webView(
+    _ webView: WKWebView,
+    decidePolicyFor navigationAction: WKNavigationAction,
+    preferences: WKWebpagePreferences,
+    decisionHandler: @escaping (WKNavigationActionPolicy, WKWebpagePreferences) -> Void
+  ) {
+    original?.webView?(
+      webView,
+      decidePolicyFor: navigationAction,
+      preferences: preferences,
+      decisionHandler: { [weak self] policy, prefs in
+        decisionHandler(policy, prefs)
+        if policy == .cancel {
+          self?.finish()
+        }
+      }
+    )
+  }
+
+  func webView(
+    _ webView: WKWebView,
+    decidePolicyFor navigationAction: WKNavigationAction,
+    decisionHandler: @escaping (WKNavigationActionPolicy) -> Void
+  ) {
+    original?.webView?(webView, decidePolicyFor: navigationAction, decisionHandler: { [weak self] policy in
+      decisionHandler(policy)
+      if policy == .cancel {
+        self?.finish()
+      }
+    })
+  }
+
+  func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+    original?.webView?(webView, didFinish: navigation)
+    webView.evaluateJavaScript("document.readyState") { [weak self] _, _ in
```

Comment:
I copy-pasted this from the original delegate. Not sure why we would execute that javascript, but then do nothing with the result. Do we execute it because webkit waits with executing it until the page is loaded?